clang/lib/DPCT/UserDefinedRules/PatternRewriter.cpp

```diff
@@ -280,8 +281,10 @@ static int parseCodeElement(const MatchPattern &Suffix,
     }
 
     const auto Character = Input[Index];
-    if(Suffix.size() == 0 && Character =='"') {
-      return Index;
+    if (SrcFileType != SourceFileType::SFT_PySetupScript) {
```

Comment:
Pls explain why  line285~line287 are skipped for SourceFileType::SFT_PySetupScript?

Reply:
Because that condition results in a failure to parse file names for Python syntax migration scenario
`f(["a.cu"])`

Where, parsing [...] in PatternRewriter results in parseCodeElement() call with empty suffix when the next element is \"

Reply:
If possible, can you please advice any other better way to do it
Or explain more about that code block